### PR TITLE
Add a description for the $referrerpolicy modifier

### PR DESCRIPTION
**Pull request by ngorskikh**

### CODE REVIEW

File: docs/general/ad-filtering/create-own-filters.md [hunk 1]
```diff
@@ -1646,6 +1647,22 @@ Basic URL exceptions shall not disable rules with `$jsonprune` modifier. They ca
 
 </details>
 
+#### **`$referrerpolicy`** {#referrerpolicy-modifier}
```
Comment (sxgunchenko) [inline docs/general/ad-filtering/create-own-filters.md:1650]: надо добавить, что можно использовать только в доверенных фильтрах, как, например, тут https://adguardteam.github.io/KnowledgeBase/general/ad-filtering/create-own-filters/#replace-modifier

**Review by sxgunchenko — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 2]
```diff
@@ -1646,6 +1647,22 @@ Basic URL exceptions shall not disable rules with `$jsonprune` modifier. They ca
...
+
+An exception rule with a modifier value disables the blocking rule with the same modifier value.
+An exception rule without a modifier value disables all matched referrer-policy rules.
+
+If a request matches multiple `$referrerpolicy` rules not disabled by exceptions, only one of them (it is unspecified, which one) is applied.
+
+**Examples**
+* `||example.com^$referrerpolicy=unsafe-url` overrides the referrer policy for `example.com` with `unsafe-url`.
+* `@@||example.com^$referrerpolicy=unsafe-url` disables the previous rule.
+* `@@||example.com/abcd.html^$referrerpolicy` disables all `$referrerpolicy` rules on `example.com/abcd.html`.
```
[the remark this answers is not in the document]
Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md:1664]: Probably depends on which URLs exactly you want to match. I just added `/abcd.html` for some variety in the examples. In any case, it has nothing to do with the modifier logic :)

File: docs/general/ad-filtering/create-own-filters.md [hunk 3]
```diff
@@ -1646,6 +1647,22 @@ Basic URL exceptions shall not disable rules with `$jsonprune` modifier. They ca
...
+#### **`$referrerpolicy`** {#referrerpolicy-modifier}
+
+These rules allow overriding of a page's [referrer policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy).
+Responses to matching requests will have all of their `Referrer-Policy` headers replaced with a single header with the value equal to the matching rule's modifier value.
+If the response carries an HTML document with a `<meta name="referrer"...` tag, the `content` attribute of the tag will also be replaced with the modifier value.
+
+An exception rule with a modifier value disables the blocking rule with the same modifier value.
+An exception rule without a modifier value disables all matched referrer-policy rules.
+
+If a request matches multiple `$referrerpolicy` rules not disabled by exceptions, only one of them (it is unspecified, which one) is applied.
```
[the remark this answers is not in the document]
Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md:1659]: Yes, the order of rules is not preserved in any way after the rules are loaded. Hence, there's no way to meaningfully combine these particular rules: the policy _does_ depend on the order (the last policy understood by the browser is applied), but we have no way to establish order between the rules. So we just apply only one of them. It is the responsibility of the user to make sure that there aren't multiple rules matching the same request :)

**Review by sxgunchenko — APPROVED** [review]

**Review by slavaleleka — APPROVED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 4]
```diff
@@ -1646,6 +1647,26 @@ Basic URL exceptions shall not disable rules with `$jsonprune` modifier. They ca
...
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:1659]:
```suggestion
If a request matches multiple `$referrerpolicy` rules not disabled by exceptions, only one of them (it is not specified which one) is applied.
```

**Review by ameshkov — CHANGES_REQUESTED** [review]
You should also explain that this modifier is only available in the apps that use CoreLibs 1.12 or newer

**Review by Alex-302 — APPROVED** [review]

**Review by sxgunchenko — APPROVED** [review]

**Review by el-termikael — APPROVED** [review]

**Review by sxgunchenko — CHANGES_REQUESTED** [review]
Please take into account changes in the recently merged PR: https://github.com/AdguardTeam/KnowledgeBase/pull/187

**Review by ameshkov — APPROVED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 5]
```diff
@@ -803,9 +803,10 @@ Disables all generic [cosmetic rules](#cosmetic-rules) on pages that correspond
 * [`$permissions`](#permissions-modifier)
 * [`$redirect`](#redirect-modifier)
 * [`$redirect-rule`](#redirect-rule-modifier)
-* [`$replace`](#replace-modifier)
```
Comment (sxgunchenko) [inline docs/general/ad-filtering/create-own-filters.md:806]: oops

**Review by sxgunchenko — APPROVED** [review]

**Review by slavaleleka — APPROVED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 6]
```diff
@@ -1417,6 +1417,8 @@ If a request matches multiple `$referrerpolicy` rules not disabled by exceptions
 * `@@||example.com^$referrerpolicy=unsafe-url` disables the previous rule.
 * `@@||example.com/abcd.html^$referrerpolicy` disables all `$referrerpolicy` rules on `example.com/abcd.html`.
 
+> `$referrerpolicy` rules are compatible only with `$document` and `$subdocument` [content type modifiers](#content-type-modifiers).
```
Comment (sxgunchenko) [inline docs/general/ad-filtering/create-own-filters.md:1420]: usually it looks like
```suggestion
> **Compatibility with other modifiers**
>
> `$referrerpolicy` rules are compatible only with `$document` and `$subdocument` [content type modifiers](#content-type-modifiers).
```

**Review by sxgunchenko — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md (hunk 6 above)
Comment (sxgunchenko) [reply docs/general/ad-filtering/create-own-filters.md:1420]: Another option is to place it under `restrictions`, but `Compatibility with other modifiers` looks like more common way :)

**Review by sxgunchenko — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md (hunk 5 above)
Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md:806]: Why? I moved them to be in alphabetical order:)

Comment (sxgunchenko) [reply docs/general/ad-filtering/create-own-filters.md:806]: I mean that I didn't sort them correctly :)

**Review by sxgunchenko — APPROVED** [review]

**Review by sxgunchenko — APPROVED** [review]

**Pull request merged**